src/r1cs/prover.rs

```diff
@@ -101,13 +109,13 @@ impl<'t, 'g> ConstraintSystem for Prover<'t, 'g> {
         let o = l * r;
 
```

Comment:
Here (and in some other places) It's possible to extract fields of `Secrets` to remove `self.secrets.<...>` boilerplate.
```rust
let Secrets { a_L, a_R, a_O, .. } = &mut self.secrets
```
Mb this will make the code a bit cleaner